Update amcrest doc for new "ptz_control" service

GaryOkie opened the pull request:
included full description of ptz_control service and its attributes along with a complete picture-elements card example showing functional arrow movement and zoom icons. 

Also mentioned this component can be used for Dahua cameras.

## Proposed change
<!-- 
    Describe the big picture of your changes here to communicate to the
    maintainers why we should accept this pull request. If it fixes a bug
    or resolves a feature request, be sure to link to that issue in the 
    additional information section.
-->



## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing (Dahua) information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->
![amcrest_ptz](https://user-images.githubusercontent.com/37629938/76991845-4a020780-6918-11ea-8ad3-aff7f3a96c9e.JPG)

- Link to parent pull request in the codebase:
https://github.com/home-assistant/core/pull/32839

- This PR fixes or closes issue:  N/A

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [ ] This PR uses the correct branch, based on one of the following:
I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

GaryOkie committed:
Update amcrest doc for new "ptz_control" service

included full description of ptz_control service and its attributes along with a complete picture-elements card example showing functional arrow movement icons. 

Also mentioned this component can be used for Dahua cameras.

frenck commented on source/_integrations/amcrest.markdown:
```diff
@@ -205,6 +206,132 @@ Service data attribute | Optional | Description
+#### Service `ptz_control`
+
+If your Amcrest or Dahua camera supports PT/PTZ, you will be able to pan, tilt or zoom your camera.  Zoom is only supported on PTZ
+cameras with optical zoom.  It does not support zooming of Varifocal lenses.
```
```suggestion
If your Amcrest or Dahua camera supports PT/PTZ, you will be able to pan, tilt or zoom your camera. Zoom is only supported on PTZ
cameras with optical zoom. It does not support zooming of Varifocal lenses.
```

frenck commented on source/_integrations/amcrest.markdown:
```diff
@@ -205,6 +206,132 @@ Service data attribute | Optional | Description
+| Service data attribute | Description |
+| -----------------------| ----------- |
+| `entity_id` | String or list of strings that point at `entity_id`s of cameras. Use `entity_id: all` to target all. |
+| `movement` | 	Direction of the movement. Allowed values: `zoom_in`, `zoom_out`, `up`, `down`, `left`, `right`, `right_up/top_right`, `right_down/bottom_right`, `left_up/top_left`,  `left_down/bottom_left` |
```
```suggestion
| `movement` | Direction of the movement. Allowed values: `zoom_in`, `zoom_out`, `up`, `down`, `left`, `right`, `right_up/top_right`, `right_down/bottom_right`, `left_up/top_left`,  `left_down/bottom_left` |
```

frenck commented on source/_integrations/amcrest.markdown:
```diff
@@ -205,6 +206,132 @@ Service data attribute | Optional | Description
+| -----------------------| ----------- |
+| `entity_id` | String or list of strings that point at `entity_id`s of cameras. Use `entity_id: all` to target all. |
+| `movement` | 	Direction of the movement. Allowed values: `zoom_in`, `zoom_out`, `up`, `down`, `left`, `right`, `right_up/top_right`, `right_down/bottom_right`, `left_up/top_left`,  `left_down/bottom_left` |
+| `travel_time` | (Optional) Camera movement start/stop time in fractional seconds. Allowed values: 0 to 1. Default:  0.2 |
```
```suggestion
| `travel_time` | (Optional) Camera movement start/stop time in fractional seconds. Allowed values: `0` to `1`. Default: `0.2`. |
```

frenck commented on source/_integrations/amcrest.markdown:
```diff
@@ -205,6 +206,132 @@ Service data attribute | Optional | Description
+</p>
+
+
+Using the following picture-elements card code you can display a live video feed from an Amcrest camera with controls for moving the camera. Note that there can be several seconds of lag before video feed reflects camera movement.   
```
```suggestion
Using the following picture elements card code, you can display a live video feed from an Amcrest camera with controls for moving the camera. Note that there can be several seconds of lag before the video feed reflects the camera movement.
```

frenck commented on source/_integrations/amcrest.markdown:
````diff
@@ -205,6 +206,132 @@ Service data attribute | Optional | Description
+        movement: right_down
+```
+
+
````
```suggestion

```

GaryOkie replied:
Happy with the formatting updates @frenck - thanks!     Please ensure this is merged coincident with the actual amcrest component merge by @pnbruckner .

frenck replied:
Please adjust this PR, if a PR collides later, we can always rebase and resolve merge conflicts.

pnbruckner replied:
I think @GaryOkie meant this doc PR shouldn't be merged until the corresponding code PR is merged. Gary, don't worry, that's SOP.

frenck reviewed (changes requested):
Thanks for the PR @GaryOkie!

I've reviewed it and left some suggestions. Could you please take a look? Thanks! 👍

GaryOkie committed:
Update source/_integrations/amcrest.markdown

Co-Authored-By: Franck Nijhof <frenck@frenck.nl>

GaryOkie committed:
Update source/_integrations/amcrest.markdown

Co-Authored-By: Franck Nijhof <frenck@frenck.nl>

GaryOkie committed:
Update source/_integrations/amcrest.markdown

Co-Authored-By: Franck Nijhof <frenck@frenck.nl>

GaryOkie committed:
Update source/_integrations/amcrest.markdown

Co-Authored-By: Franck Nijhof <frenck@frenck.nl>

GaryOkie committed:
Update source/_integrations/amcrest.markdown

Co-Authored-By: Franck Nijhof <frenck@frenck.nl>

GaryOkie committed:
removed "top/bottom" alternate movement commands.

pnbruckner commented on source/_integrations/amcrest.markdown:
```diff
@@ -153,7 +153,8 @@ Available services:
 `goto_preset`, `set_color_bw`,
-`start_tour` and `stop_tour`
+`start_tour` and `stop_tour`,
+`ptz_control`
```
Shouldn't "and" be between `stop_tour` & `ptz_control`?

GaryOkie replied:
yep

pnbruckner commented on source/_integrations/amcrest.markdown:
```diff
@@ -205,6 +206,129 @@ Service data attribute | Optional | Description
+#### Service `ptz_control`
+
+If your Amcrest or Dahua camera supports PT/PTZ, you will be able to pan, tilt or zoom your camera. Zoom is only supported on PTZ
+cameras with optical zoom. It does not support zooming of Varifocal lenses.
```
> Zoom is only supported on PTZ cameras with optical zoom. It does not support zooming of Varifocal lenses.

I'm not sure this is true. I tested this on a PTZ camera that I have that only has digital zoom and it works fine. I would recommend removing these sentences.

GaryOkie replied:
You're right about zoom working fine for a PTZ camera with digital zoom only (not just optical).   In my earlier testing it didn't work from my PTZ camera because of the original error with zoom-in/out reversed!  

However, we should still advise this zoom does not work for non-PTZ cameras with VariFocal lenses.   Often a VariFocal is a "set and forget" for the desired image.   I have a couple of them and rarely  use the zoom after setup, but if we were to add this capability, it would require an update to python-amcrest.

```
http://<ip>/cgi-bin/devVideoInput.cgi?action= adjustFocus&focus=<focus>&zoom=<zoom>
   focus: float, the range is between 0 and 1; -1 means reset to position 0.
   zoom: float, the range is between 0 and 1; -1 means reset to position 0.
```

I have no idea how to determine what the Focus values should be, but I have deferred to AutoFocus after any attempt to zoom.   I have tested this manually passing these http commands, and it works fine.  I doubt this is worth the effort to include.

pnbruckner replied:
> However, we should still advise this zoom does not work for non-PTZ cameras with VariFocal lenses.

Um, it's a PTZ command, so why would you have to warn it doesn't work with non-PTZ cameras? It doesn't work with cows either. Should we warn about that? :stuck_out_tongue_winking_eye: But seriously, I don't really care either way.

GaryOkie replied:
I think it would be reasonable for someone to "try it and see" if they have a camera capable of optical  **zooming** but not pan and tilt.   I know I would!   Just saving some knuckleheads like me some effort is all.

pnbruckner commented on source/_integrations/amcrest.markdown:
```diff
@@ -205,6 +206,129 @@ Service data attribute | Optional | Description
+
+| Service data attribute | Description |
+| -----------------------| ----------- |
+| `entity_id` | String or list of strings that point at `entity_id`s of cameras. Use `entity_id: all` to target all. |
```
This should be consistent with the `entity_id` description of all the other services. Either change them all or make this one the same as the others. It would be nice if you changed them all to:

The entity ID of the camera to control. May be a list of multiple entity IDs. To target all cameras, set entity ID to `all`.

GaryOkie replied:
changed them all to your suggestion.

pnbruckner commented on source/_integrations/amcrest.markdown:
```diff
@@ -205,6 +206,129 @@ Service data attribute | Optional | Description
+### Example card with controls
+
+<p class='img'>
+  <img src='/images/integrations/amcrest/amcrest_ptz.jpg' alt='Screenshot using a picture-elements with PTZ controls.'>
```
You need to add the picture file.

GaryOkie replied:
I did the "Upload file" with a new "amcrest_ptz.jpg" image from the main page which I think was a mistake, but I don't see it listed to remove it.  
 
If I navigate to https://github.com/GaryOkie/home-assistant.github.io/tree/patch-1 , I can then navigate to /source/images/integrations .    There isn't a create folder option I could see, but I tried create file, hoping it would give me a folder option.  Nope, now I've got an empty amcrest file that needs to be removed.

EDIT:  Nevermind - figured it out.   Image uploaded to new amcrest folder and empty file removed.    Still not sure about where that initial uploaded image went or if it will cause an issue.

Lastly, when reviewing the rendered markdown, why is the description for ptz_control entity_id not displaying?   I can't see anything wrong.

pnbruckner replied:
First, the picture still isn't right. You have two files in the amcrest folder, one named .JPG and the other .jpg. I think you need to delete the .jpg one and rename the other from .JPG to .jpg, at the very least. I don't know if the `<img src=...>` link is right or not.

Regarding the ptz_control entity_id row, I think that's because you have:
```
| Service data attribute | Description |
| -----------------------| ----------- |
```
instead of:
```
Service data attribute | Optional | Description
-|-|-
```
Look at the other service descriptions as an example.

pnbruckner replied:
> Still not sure about where that initial uploaded image went or if it will cause an issue.

Looks like you committed it to the `next` branch instead of your `patch-1` branch.

GaryOkie replied:
thanks.  The markdown is fixed now.   fixing the image file name now...

EDIT:  Got the new image file uploaded properly.    Other JPG file in the next branch removed.

GaryOkie committed:
implement Phil's corrections/suggestions

Also updated the picture-elements card to add a zoom-in/out icon in the center of the arrows.

New example image will also be uploaded once I figure out how to do that in the markdown.

GaryOkie committed:
attempt to create an amcrest folder (for images)

GaryOkie committed:
delete empty amcrest file.

It needed to be a folder.
yet to figure out how to make it so.

GaryOkie committed:
create /amcrest/amcrest_ptz.jpg 

create /amcrest/dummy file in the hopes this creates a new folder

GaryOkie committed:
Upload new amcrest_ptz image

finally figured out how to create a new folder in multiple steps.

GaryOkie committed:
fix ptz_control table formatting

Problem was copying the ptz control markdown from Foscam doc which had 2 columns, not 3.
(all this -|-|- lingo was new to me, and thought copying what worked for one, would work for the other)

Also moved the 2 usage notes to their own subsection

GaryOkie committed:
Delete amcrest_ptz.JPG

deleting because extension is all caps, not lc.    did not see a way to rename a committed file.

GaryOkie committed:
amcrest_ptz.jpg uploaded

frenck committed:
:pencil2: Tweak

frenck reviewed (approved):
Thanks, @GaryOkie! 👍
